Google.Api.Generator/Generation/MetadataGenerator.cs

```diff
@@ -0,0 +1,84 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Gapic.Metadata;
+using static Google.Gapic.Metadata.GapicMetadata.Types;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Google.Api.Generator.Generation
+{
+    internal static class MetadataGenerator
+    {
+        private const string GapicMetadataSchemaVersion = "1.0";
+        private const string GapicMetadataFileDescription = "This file maps proto services/RPCs to the corresponding library clients/methods";
+        private const string GapicMetadataLanguage = "csharp";
+
+        private const string TransportKeyGrpc = "grpc";
+
+        internal static string GenerateGapicMetadataJson(List<ServiceDetails> allServiceDetails)
+        {
+            var gapicMetadata = new GapicMetadata
+            {
+                Schema = GapicMetadataSchemaVersion,
+                Comment = GapicMetadataFileDescription,
+                Language = GapicMetadataLanguage,
+                ProtoPackage = allServiceDetails.First().ProtoPackage,
+                LibraryPackage = allServiceDetails.First().Namespace,
+                Services = {
+                allServiceDetails.ToDictionary(
+                    serviceDetails => serviceDetails.ServiceName,
+                    ServiceForTransportMetadata
+                )}
+            };
+
+            JObject obj = JsonConvert.DeserializeObject<JObject>(
+                gapicMetadata.ToString(),
+                new JsonSerializerSettings { DateParseHandling = DateParseHandling.None }
+            );
+
+            if (obj == null)
+            {
+                throw new InvalidOperationException("Json conversion failed for the Gapic Metadata object.");
+            }
+
+            return obj.ToString(Formatting.Indented) + "\n"; //trailing file newline to please github;
+        }
+
+        private static ServiceForTransport ServiceForTransportMetadata(ServiceDetails serviceDetails) =>
+            new ServiceForTransport
+            {
+                Clients = { new Dictionary<string, ServiceAsClient>
+                {
+                    { 
+                        TransportKeyGrpc,
+                        new ServiceAsClient
+                        {
+                            LibraryClient = serviceDetails.ClientImplTyp.Name,
+                            Rpcs = { serviceDetails.Methods.ToDictionary(
+                                methodDetails => methodDetails.ProtoRpcName,
+                                methodDetails => new MethodList
+                                {
+                                    Methods = { methodDetails.SyncMethodName, methodDetails.AsyncMethodName }
+                                }
+                            )}
+                        }
+                    }
+                }}
+            };

```

Comment:
nits on curly brackets possitions, etc.
```suggestion
            new ServiceForTransport
            {
                Clients = 
                { 
                    new Dictionary<string, ServiceAsClient>
                    {
                        { 
                            TransportKeyGrpc,
                            new ServiceAsClient
                            {
                                LibraryClient = serviceDetails.ClientImplTyp.Name,
                                Rpcs = 
                                { 
                                    serviceDetails.Methods.ToDictionary(
                                    methodDetails => methodDetails.ProtoRpcName,
                                    methodDetails => new MethodList
                                    {
                                        Methods = { methodDetails.SyncMethodName, methodDetails.AsyncMethodName }
                                    })
                                }
                            }
                        }
                    }
                }
            };
```

Reply:
👍  but at this point we should either have a linting engine in-place or not be strict on bracket placement